plugin/builtin/patchtransformer/PatchTransformer_test.go

```diff
@@ -746,8 +743,72 @@ spec:
       - image: test-image
         name: test-deployment
         ports:
+        - containerPort: 8080
+          name: take-over-the-world
+          protocol: TCP
         - containerPort: 8080
           name: disappearing-act
           protocol: TCP
 `)
 }
+
+func TestPatchTransformerSecondaryMergeKey(t *testing.T) {
+	th := kusttest_test.MakeEnhancedHarness(t).
+		PrepBuiltin("PatchTransformer")
+	defer th.Reset()
+
+	th.RunTransformerAndCheckResult(`
+apiVersion: builtin
+kind: PatchTransformer
+metadata:
+  name: test-transformer
+patch: |-
+  apiVersion: apps/v1
+  kind: Deployment
+  metadata:
+    name: test-transformer
+    labels:
+      test-transformer: did-my-job
+  spec:
+    template:
+      spec:
+        containers:
+        - image: test-image
+          name: test-deployment
+          ports:
+          - containerPort: 8080
+            protocol: different
+target:
+  kind: Deployment
+  name: test-deployment
+`, `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: test-deployment
+spec:
+  template:
+    spec:
+      containers:
+      - image: test-image
+        name: test-deployment
+        ports:
+        - containerPort: 8080
+          protocol: TCP
+`, `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    test-transformer: did-my-job
+  name: test-deployment
+spec:
+  template:
+    spec:
+      containers:
+      - image: test-image
+        name: test-deployment
+        ports:
+        - containerPort: 8080
+          protocol: different
```

Comment:
is this correct? Shouldn't you have both?

Reply:
Some context: this is the expected behavior prior to this PR. There was another PR I made to fix #2767 that caused a test to fail in kpt, so I added this test (similar to the kpt one) to make sure my changes here wouldn't break it. 

Basically, if there is no name field, then ports should merge on the containerPort field. (#3111)

@monopole Can you confirm if I've understood this correctly?

Reply:
just to close this - we're not going to use `name` now ( https://github.com/kubernetes-sigs/kustomize/pull/3125#discussion_r512155545)

Reply:
Thanks Jeff, I think that's the right thing to do!